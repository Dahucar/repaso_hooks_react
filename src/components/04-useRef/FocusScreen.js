import React, { useRef } from 'react'

export const FocusScreen = () => {

    // Lo usariamos para crear referencias mutables
    const inputRef = useRef();
    
    const handleCLick = () => {
        //de menera tradicional
        //document.querySelector('input').select();
        //usando un hook
        inputRef.current.select();
        //inputRef guardara la referencia del elemento html
        console.log(inputRef);
    }

    return (
        <div className="container card">
            <div className="card-header">
                <h2>Focus Screend</h2>
            </div>
            <div className="card-body">
                <input 
                    ref={ inputRef }
                    className="form-control"
                    placeholder="Ingrese nombre"
                />
            </div>
            <div className="card-footer">
                <button className="btn btn-outline-success"
                    onClick={ handleCLick }>
                    Click
                </button>
            </div>
        </div>
    )
}
