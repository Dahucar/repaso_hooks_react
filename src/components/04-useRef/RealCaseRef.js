import React, { useState } from 'react'
import { MultipleCustomHooks } from '../03-examples/MultipleCustomHooks'

export const RealCaseRef = () => {

    // Caso de real para uso de hook useRef: Mostrar componente de manera condicional.

    const [show, setShow] = useState(false);

    return (
        <div className="container card">
            <div className="card-header">
                <h2>RealCaseRef</h2>
            </div>
            <div className="card-body">
                { show && <MultipleCustomHooks />}
            </div>
            <div className="card-footer">
                <button 
                    type="button" 
                    className="btn btn-success btn-lg btn-block"
                    onClick={ () => { setShow(!show) } }
                >
                    { show ? 'Ocultar' : 'Mostrar' }
                </button>
            </div>
        </div>
    )
}
