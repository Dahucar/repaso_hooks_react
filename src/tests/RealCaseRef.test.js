import { shallow } from 'enzyme';
import { RealCaseRef } from '../components/04-useRef/RealCaseRef';

describe('Pruebas en <RealCaseRef />', () => {
    const wrapper = shallow( <RealCaseRef /> );
    test('Debe de hacer match con su instantanea.', () => {
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('MultipleCustomHooks').exists() ).toBe( false );
    });
    test('Debe de mostrar el componente <MultipleCustomHooks/>', () => {
        wrapper.find( 'button' ).simulate('click');
        expect( wrapper.find('MultipleCustomHooks').exists() ).toBe( true );
    });
});
