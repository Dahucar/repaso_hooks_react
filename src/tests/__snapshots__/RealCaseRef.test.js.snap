// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <RealCaseRef /> Debe de hacer match con su instantanea. 1`] = `
<div
  className="container card"
>
  <div
    className="card-header"
  >
    <h2>
      RealCaseRef
    </h2>
  </div>
  <div
    className="card-body"
  />
  <div
    className="card-footer"
  >
    <button
      className="btn btn-success btn-lg btn-block"
      onClick={[Function]}
      type="button"
    >
      Mostrar
    </button>
  </div>
</div>
`;
